### PR TITLE
IMP allow custom maven repositories

### PR DESCRIPTION
**Pull request by skliarpawlo**

### CODE REVIEW

**Comment by drudim** [comment]
Could we do the same with `spark.jars.ivy`?

> Comma-separated list of additional remote repositories to search for the coordinates given with spark.jars.packages.

However it could be local `ivy`, not sure.

**Comment by skliarpawlo** [comment]
@drudim https://issues.apache.org/jira/browse/SPARK-19630
That is confusing doc which is fixed in newer releases
>  Path to specify the Ivy user directory, used for the local Ivy cache and package files from spark.jars.packages. This will override the Ivy property ivy.default.ivy.user.dir which defaults to ~/.ivy2.

File: tests/unit/test_session.py [hunk 1]
```diff
@@ -81,7 +101,7 @@ class _Session(SparklySession):
 
         self.assertEqual(os_mock.environ, {
             'PYSPARK_PYTHON': sys.executable,
-            'PYSPARK_SUBMIT_ARGS': ' --jars file_a.jar,file_b.jar pyspark-shell',
+            'PYSPARK_SUBMIT_ARGS': '  --jars file_a.jar,file_b.jar pyspark-shell',
```
Comment (drudim) [inline tests/unit/test_session.py]: if it's possible, let's normalise string so we don't need to test 2 vs 1 whitespace

**Comment by skliarpawlo** [comment]
@drudim fixed, ready for review

**Comment by drudim** [comment]
Probably forgot to submit during last review: can we update documentation? https://sparkly.readthedocs.io/en/latest/session.html

**Comment by skliarpawlo** [comment]
@drudim  fixed

File: docs/source/session.rst [hunk 2]
```diff
@@ -66,6 +66,26 @@ declared as part of the job.
     df = spark.read_ext.by_url('cassandra://<cassandra-host>/<db>/<table>?consistency=QUORUM')
 
 
+Custom Maven repositories
+-------------------------
+
+**Why**: If you have private maven repository, this is the way how to point spark to it to make package lookup.
```
Comment (drudim) [inline docs/source/session.rst]: "If you have a private maven repository"

File: docs/source/session.rst [hunk 3]
```diff
@@ -66,6 +66,26 @@ declared as part of the job.
...
+
+**For example**: Let's assume your maven repository is available on: http://my.repo.net/maven,
```
Comment (drudim) [inline docs/source/session.rst]:
```
 --packages                  Comma-separated list of maven coordinates of jars to include
                              on the driver and executor classpaths. Will search the local
                              maven repo, then maven central and any additional remote
                              repositories given by --repositories. The format for the
                              coordinates should be groupId:artifactId:version.
  --exclude-packages          Comma-separated list of groupId:artifactId, to exclude while
                              resolving the dependencies provided in --packages to avoid
                              dependency conflicts.
  --repositories              Comma-separated list of additional remote repositories to
                              search for the maven coordinates given with --packages.
```
Trying to understand the order:
- local maven (local fs)
- maven central
- packages
Correct?

**Comment by skliarpawlo** [comment]
@drudim can you pls fix the doc to your liking and let's merge this

**Comment by drudim** [comment]
@skliarpawlo I am ok to merge this as soon as you answer https://github.com/Tubular/sparkly/pull/44#discussion_r123297474
I would like us to put this answer to the doc.

**Comment by skliarpawlo** [comment]
Quite hard to be sure what the precedence is but from this two sources I conclude that it is: local, custom, maven-central:
   1. https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala#L1088 
 2. https://issues.apache.org/jira/browse/SPARK-17568

Added this to docs
